unit_testing/cli/test_cli_sct_smooth_spinalcord.py

```diff
@@ -1,19 +1,18 @@
-from pytest_console_scripts import script_runner
 import pytest
 import logging
 import os
+
 from spinalcordtoolbox.scripts import sct_smooth_spinalcord
 
 logger = logging.getLogger(__name__)
 
 
-@pytest.mark.script_launch_mode('subprocess')
-def test_sct_smooth_spinalcord_backwards_compat(script_runner):
-    ret = script_runner.run('sct_testing', '--function', 'sct_smooth_spinalcord')
-    logger.debug(f"{ret.stdout}")
-    logger.debug(f"{ret.stderr}")
-    assert ret.success
-    assert ret.stderr == ''
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+def test_sct_smooth_spinalcord_no_checks():
+    """Run the CLI script without checking results.
+    TODO: Check the results. (This test replaces the 'sct_testing' test, which did not implement any checks.)"""
+    sct_smooth_spinalcord.main(argv=['-i', 't2/t2.nii.gz', '-s', 't2/t2_seg-manual.nii.gz', '-smooth', '0,0,5'])
 
 
 def test_sct_smooth_spinalcord_o_flag(tmp_path):
```

Comment:
Can you delete the `os.unlink`s in this? I guess that `@pytest.mark.usefixtures("run_in_sct_testing_data_dir")` still isn't `tmp_path`, so the output will still muck up the input no matter what :/

Reply:
You're right... The fact that `os.unlinks` are necessary is a good smell that `sct_smooth_spinalcord` could use an `-ofolder` argument. Right now, it pollutes the working directory no matter what we do.